src/MSBuild/TerminalLogger/TerminalLogger.cs

```diff
@@ -25,6 +25,9 @@ namespace Microsoft.Build.Logging.TerminalLogger;
 /// </remarks>
 internal sealed class TerminalLogger : INodeLogger
 {
+    private const string FilePathPattern = " -> ";
+    private readonly string[] _immediateMessageMarkers = new[] { "[CredentialProvider]", "--interactive" };
```

Comment:
Can you comment these? Is it that if you're not in interactive, the message will say "add --interactive`, and if you are in interactive, the message has `[CredentialProvider]`?

Reply:
"add --interactive" falls in `WarningRaised`, and "[CredentialProvider]" goes to `MessageRaised`.
If any of these substrings appears we put it in output immediately:
![image](https://github.com/dotnet/msbuild/assets/95473390/47f608e8-eb0a-464c-a572-98f1937328da)
![image](https://github.com/dotnet/msbuild/assets/95473390/eed5480f-d271-420b-8fbe-03b745f6e11f)

Reply:
Man I hate that formatting, but it's probably the best we can reasonably do given the free-form nature of the Messages. My ideal long-term fix for this would have different formatting for sure :)

Side question - do more advanced terminals (iTerm2, Windows Terminal, etc?) make the devicelogin url clickable?

Reply:
Windows Terminal +
iTerm2 -
bash -

Reply:
@baronfel , I removed the coloring, let me know if you desperately want it back ;)